packages/grafana-ui/src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,52 @@
+import React, { CSSProperties, FC, ReactNode } from 'react';
+import RcDrawer from 'rc-drawer';
+import { css, cx } from 'emotion';
+import { getTheme, stylesFactory } from '../../themes';
+import { GrafanaTheme } from '../../types';
+
+interface Props {
+  children: ReactNode;
+  /** Title shown at the top of the drawer */
+  title?: string;
+  maskClosable?: boolean;
+  inline?: boolean;
+  width?: number;
+
+  onClose: () => void;
+}
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+  titleContainer: css`
+    font-size: ${theme.typography.size.lg};
+    display: flex;
+    justify-content: space-between;
+    align-items: center;
+  `,
+  close: css`
+    cursor: pointer;
+  `,
+}));
+
+export const Drawer: FC<Props> = ({ children, inline = false, onClose, maskClosable = false, title, width = 30 }) => {
+  const theme = getTheme();
+  const drawerStyles = getStyles(theme);
+
+  return (
+    <RcDrawer
+      onClose={onClose}
+      handler={false}
+      open={true}
```

Comment:
I like this. Thanks to that we give the control of whether the drawer is rendered at all to the the component that's using it.